src/mrinufft/operators/interfaces/nudft_numpy.py

```diff
@@ -4,23 +4,42 @@
 
 import numpy as np
 import scipy as sp
-
 from ..base import FourierOperatorCPU
-from mrinufft._utils import proper_trajectory
+from mrinufft._utils import proper_trajectory, get_array_module
 
 
 def get_fourier_matrix(ktraj, shape, dtype=np.complex64, normalize=False):
     """Get the NDFT Fourier Matrix."""
+    module = get_array_module(ktraj)
     ktraj = proper_trajectory(ktraj, normalize="unit")
     n = np.prod(shape)
     ndim = len(shape)
-    matrix = np.zeros((len(ktraj), n), dtype=dtype)
-    r = [np.linspace(-s / 2, s / 2 - 1, s) for s in shape]
-    grid_r = np.reshape(np.meshgrid(*r, indexing="ij"), (ndim, np.prod(shape)))
-    traj_grid = ktraj @ grid_r
-    matrix = np.exp(-2j * np.pi * traj_grid, dtype=dtype)
+
+    if module.__name__ == "torch":
+        torch = module
+        device = ktraj.device
+        dtype = torch.complex64
+        r = [torch.linspace(-s / 2, s / 2 - 1, s, device=device) for s in shape]
+        grid_r = torch.meshgrid(r, indexing="ij")
+        grid_r = torch.reshape(torch.stack(grid_r), (ndim, n)).to(device)
+        traj_grid = torch.matmul(ktraj, grid_r)
+        matrix = torch.exp(-2j * np.pi * traj_grid).to(dtype).to(device).clone()
+    else:
+        r = [np.linspace(-s / 2, s / 2 - 1, s) for s in shape]
+        grid_r = np.reshape(np.meshgrid(*r, indexing="ij"), (ndim, np.prod(shape)))
+        traj_grid = ktraj @ grid_r
+        matrix = np.exp(-2j * np.pi * traj_grid, dtype=dtype)
```

Comment:
This can be refactored more:

To make it cleaner, maybe have `xp = get_array_module`, then use xp.

```python
    r = [module.linspace(-s / 2, s / 2 - 1, s, dtype=dtype) for s in shape]
    grid_r = module.meshgrid(r, indexing="ij")
    grid_r = module.reshape(torch.stack(grid_r), (ndim, n))
    if module.__name__ == 'torch':
        grid_r = grid_r.to(device)
    traj_grid = module.matmul(ktraj, grid_r)
    matrix = module.exp(-2j * np.pi * traj_grid)
```


Dont commit this code as is, its just a suggestion. This way we can prevent a lot of repeations

Reply:
Here, I think my code isn't concise yet, because I found that we have to put the operations to device when torch and it can't be executed when numpy...Do you have some ideas to solve this problem??

Reply:
> This can be refactored more:
> 
> To make it cleaner, maybe have `xp = get_array_module`, then use xp.
> 
> ```python
>     r = [module.linspace(-s / 2, s / 2 - 1, s, dtype=dtype) for s in shape]
>     grid_r = module.meshgrid(r, indexing="ij")
>     grid_r = module.reshape(torch.stack(grid_r), (ndim, n))
>     if module.__name__ == 'torch':
>         grid_r = grid_r.to(device)
>     traj_grid = module.matmul(ktraj, grid_r)
>     matrix = module.exp(-2j * np.pi * traj_grid)
> ```
> 
> Dont commit this code as is, its just a suggestion. This way we can prevent a lot of repeations

Here, I think my code isn't concise yet, because I found that we have to put the operations to device when torch and it can't be executed when numpy...Do you have some ideas to solve this problem??